keps/sig-network/4631-loadbalancerstatus/README.md

```diff
@@ -0,0 +1,1210 @@
+resource-provisioning step for the load balancer. When
+`LoadBalancerProvisioning` is `True`, the load balancer is either not
+handling connections, or else is still handling connections according
+to the semantics of a previous state of the Service (with the value of
+the `LoadBalancerServing` condition distinguishing these two cases).
+
+The cloud provider should set a `LoadBalancerProvisioning` condition
+immediately upon first becoming aware of a `Type: LoadBalancer`
+Service. At all times, the condition should be `True` if the cloud
+provider is waiting for the completion of an operation to bring the
+load balancer's behavior in line with the current state of the Service
+object, and `False` if it not. (There are no defined semantics for an
+`Unknown` value of this condition.)
+
+Any time any load-balancer-related field (or provider-specific
+annotation) of the Service object changes, the cloud provider must
+update at least the `LastTransitionTime` of the
+`LoadBalancerProvisioning` condition, to indicate that it has seen the
+update, even if the value of `LoadBalancerProvisioning` remains
+`False`.
+
+#### Terminating Condition
+
+```
+<<[UNRESOLVED termination ]>>
+
+Do we need/want a `LoadBalancerTerminating` condition? And/or, should
+the cloud provider set `LoadBalancerProvisioning` when it is
+*de*-provisioning a load balancer?
+
+There is possibly no need for this when deleting a Service, but it may
+help in the case of converting a Service from `Type: LoadBalancer` to
+`Type: ClusterIP` (which is tested by the e2es).
+
+<<[/UNRESOLVED]>>
+```
+
+#### Service Lifecycle and Interaction of the Conditions
+
+When a load balancer Service is first created (or a non-load-balancer
+Service is converted to `Type: LoadBalancer`), one of three things
+should immediately happen:
+
+  1. The cloud provider sets `LoadBalancerProvisioning=False`,
+     `LoadBalancerServing=False`, with appropriate `Reason` values
+     (discussed below), indicating that it cannot or will not
+     provision a load balancer for the service.
+
+  2. The cloud provider sets `LoadBalancerProvisioning=False`,
+     `LoadBalancerServing=True`, (and fills in `.Status.LoadBalancer`)
+     indicating that the load balancer for the service is already
+     available.
+
+  3. The cloud provider sets `LoadBalancerProvisioning=True`,
+     `LoadBalancerServing=False`, with appropriate `Reason` values,
+     indicating that it is provisioning the load balancer.
+
+In the third case, this would later be followed up by the cloud
+provider setting `LoadBalancerProvisioning` to `False` and
+`LoadBalancerServing` to either `True` or `False`, depending on
+whether the provisioning succeeded or failed.
+
+If the `LoadBalancerServing` and `LoadBalancerProvisioning` conditions
+do not appear on the Service within a reasonable amount of time after
+the Service is created (or modified to be `Type: LoadBalancer`), the
+user should assume that no cloud provider is running, and that
```

Comment:
Or that the cloud provider does not implement this extension.  I don't think we can say more than that.